File: design-patterns/ts/strategy/transport/car.ts
// car.ts

import Transport from "./transport";

class Car implements Transport {
    start(): void {
        console.log("Car started");
    }

    stop(): void {
        console.log("Car stopped");
    }

    repair(): void {
        console.log("Car repair");
    }

    getInfo(): void {
        console.log("Transport type: Car");
        console.log("Number of wheels: 4");
        console.log("Average Weight: 4,000 Pounds");
    }

    operate(): void {
        console.log("Driving car ............");
    }
}

export default Car;

File: design-patterns/go/strategy/transport/car.go
// car.go

package main

import "fmt"

type Car struct {
}

func NewCar() (car *Car) {
	car = &Car{}
	return
}

func (car *Car) GetInfo() {
	fmt.Println("Transport type: Car")
	fmt.Println("Number of wheels: 4")
	fmt.Println("Average Weight: 4,000 Pounds")
}

func (car *Car) Operate() {
	fmt.Println("Driving car ............")
}

func (car *Car) Repair() {
	fmt.Println("Car repair")
}

func (car *Car) Start() {
	fmt.Println("Car started")
}

func (car *Car) Stop() {
	fmt.Println("Car stopped")
}
